pkg/cmd/search/repos/repos.go

```diff
@@ -0,0 +1,238 @@
+	cmd.Flags().Var(opts.Query.Qualifiers["Visibility"], "visibility", "Filter based on visibility")
+
+	return cmd
+}
+
+func reposRun(opts *ReposOptions) error {
+	io := opts.IO
+	cfg, err := opts.Config()
+	if err != nil {
+		return err
+	}
+	host, err := cfg.DefaultHost()
+	if err != nil {
+		return err
+	}
+	client, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+	searcher := shared.NewSearcher(host, client)
+	if opts.WebMode {
+		url := searcher.URL(opts.Query)
+		if io.IsStdoutTTY() {
+			fmt.Fprintf(io.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(url))
+		}
+		return opts.Browser.Browse(url)
+	}
+	opts.IO.StartProgressIndicator()
+	result, err := searcher.Search(opts.Query)
+	opts.IO.StopProgressIndicator()
+	if err != nil {
+		return err
+	}
+	if err := opts.IO.StartPager(); err == nil {
+		defer opts.IO.StopPager()
+	} else {
+		fmt.Fprintf(opts.IO.ErrOut, "failed to start pager: %v\n", err)
+	}
+	if opts.JqExpression != "" {
+		j, err := json.Marshal(result.Items)
+		if err != nil {
+			return err
+		}
+		err = export.FilterJSON(io.Out, bytes.NewReader(j), opts.JqExpression)
+		if err != nil {
+			return err
+		}
+	} else if opts.GoTemplate != "" {
+		t := export.NewTemplate(opts.IO, opts.GoTemplate)
+		j, err := json.Marshal(result.Items)
+		if err != nil {
+			return err
+		}
+		err = t.Execute(bytes.NewReader(j))
+		if err != nil {
+			return err
+		}
+	} else {
+		err := displayResults(opts.IO, result)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func displayResults(io *iostreams.IOStreams, results search.Result) error {
+	cs := io.ColorScheme()
+	tp := utils.NewTablePrinter(io)
+	for _, repo := range results.Items {
+		var tags []string
+		private, _ := repo["private"].(bool)
+		fork, _ := repo["fork"].(bool)
+		archived, _ := repo["archived"].(bool)
+		if private {
+			tags = append(tags, "private")
+		} else {
+			tags = append(tags, "public")
+		}
+		if fork {
+			tags = append(tags, "fork")
+		}
+		if archived {
+			tags = append(tags, "archived")
+		}
+		info := strings.Join(tags, ", ")
+		infoColor := cs.Gray
+		if private {
+			infoColor = cs.Yellow
+		}
+		updatedAt, _ := repo["updated_at"].(string)
+		tp.AddField(repo["full_name"].(string), nil, cs.Bold)
+		description, _ := repo["description"].(string)
+		tp.AddField(text.ReplaceExcessiveWhitespace(description), nil, nil)
+		tp.AddField(info, nil, infoColor)
+		if tp.IsTTY() {
+			t, _ := time.Parse(time.RFC3339, updatedAt)
+			tp.AddField(t.Format(time.RFC822), nil, cs.Gray)
+		} else {
+			tp.AddField(updatedAt, nil, nil)
+		}
+		tp.EndRow()
+	}
+
+	if io.IsStdoutTTY() {
+		header := "No repositories matched your search\n"
+		if len(results.Items) > 0 {
+			header = fmt.Sprintf("Showing %d of %d repositories\n\n", len(results.Items), results.TotalCount)
+		}
+		fmt.Fprintf(io.Out, "\n%s", header)
+	}
+
+	return tp.Render()
+}
+
+func NewSearchQuery() search.Query {
```

Comment:
This is extremely tedious to have to define the flags and also the query qualifiers here. I couldn't come up with a simpler way to approach this problem. I am open to any ideas on how to make writing these less painful for future search commands.